hnn_core/gui.py

```diff
@@ -0,0 +1,1111 @@
+"""IPywidgets GUI."""
+
+# Authors: Mainak Jas <mjas@mgh.harvard.edu>
+#          Huzi Cheng <hzcheng15@icloud.com>
+
+import codecs
+import logging
+import multiprocessing
+import os
+import os.path as op
+
+import matplotlib.pyplot as plt
+import numpy as np
+from IPython.display import display
+from ipywidgets import (HTML, Accordion, AppLayout, BoundedIntText,
+                        BoundedFloatText, Button, Dropdown, FileUpload,
+                        FloatLogSlider, FloatText, GridspecLayout, HBox,
+                        IntText, Layout, Output, RadioButtons, Tab, Text, VBox,
+                        interactive_output)
+
+import hnn_core
+from hnn_core import (JoblibBackend, MPIBackend, jones_2009_model, read_params,
+                      simulate_dipole)
+from hnn_core.params import (_extract_drive_specs_from_hnn_params, _read_json,
+                             _read_legacy_params)
+from hnn_core.viz import plot_dipole
+
+
+log_file = os.getenv("HNNGUI_LOGFILE", None)
+debug_gui = os.getenv("DEBUG_HNNGUI", "0")
+if debug_gui == '1' and log_file is not None:
+    logging.basicConfig(filename=log_file,
+                        filemode='w',
+                        level=logging.DEBUG,
+                        format='%(name)s - %(levelname)s - %(message)s')
+else:
+    logging.basicConfig(level=logging.ERROR)
```

Comment:
Probably better to have this through flags in the command line rather than environment variables. E.g:

```py
$ hnn-gui --logfile log.txt
```

and just set logging level to INFO. Keep it simple

Reply:
This probably is not going to work as the `main` function will call the notebook again and all the `sys.argv` will be lost, so in the second call of `gui.py` we will have no information about the `logfile`. Since now it only involves reading environment variables rather than writing them, I don't think we really need to change this at the moment.